packages/app/src/components/actions/deposit.tsx

```diff
@@ -79,14 +77,24 @@ const Deposit: FC<Props> = ({ name, symbol, tokenBalance, tenderTokenBalance })
       </Box>
 
       <Box fill="horizontal" direction="row" justify="center" align="center">
-        <Form>
-          <FormField label="Deposit Amount" controlId="formDeposit">
+        <Form validate="change">
+          <FormField
+            fill
+            label="Deposit Amount"
+            name="depositAmount"
+            validate={() => {
+              if (isPositiveAndSmallerThanMax(depositInput, tokenBalance)) {
+                return { message: "Please provide an available amount", status: "error" };
```

Comment:
What do you think about splitting this function up ? 

Generally I prefer functions that do 1 thing. 

- isPositive (how about `isValidUint`  this implies both a valid integer as well as positive) 
- exceedsBalance 

Based on that we can display following error messages:
- "Please provide a valid amount"
- "Amount exceeds available balance"